src/components/folder/NavBox.jsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from "react";
+import { getFolderNavInfo } from "../../api/folder.js";
+import * as S from './NavBox.styled.jsx';
+import add from '../../assets/icon/add.svg';
+import NavItem from './NavItem';
+import shareIcon from '../../assets/icon/share.svg';
+import penIcon from '../../assets/icon/pen.svg';
+import deleteIcon from '../../assets/icon/delete.svg';
+
+const NavBox = ({navId, onClickNavItem}) => {
+    const [navList, setNavList] = useState([]);
+    const [currentNav, setCurrentNav] = useState('전체');
+
+    const handleLoadInfo = async () => {
+        const folderNavInfo = await getFolderNavInfo();
+
+        if(folderNavInfo !== null){
+            setNavList(folderNavInfo.data);
+        }
+    }
+
+    useEffect(() => {
+        handleLoadInfo();
+    },[]);
+
+
+    useEffect(() => {
+        if(navId === undefined){setCurrentNav('전체')}
+
+        navList.map((navListItem) => {
+            if(navListItem.id === navId) setCurrentNav(navListItem.name);
+        })
```

Comment:
map 으로 하나씩 비교할 필요 없이 find 로 원하는 것만 바로 가져오면 좋을 거 같아요!
```suggestion
        const selectedNav = navList.find((navListItem) => navListItem.id === navId)
        setCurrentNav(selectedNav? selectedNav.name :  '전체')
```